test/marshaling_test.ts

```diff
@@ -17,6 +17,7 @@ describe('Marshaling', () => {
     assert.deepEqual([1], transform([1]));
     assert.deepEqual({a: 1}, transform({a: 1}));
     assert.deepEqual(undefined, transform(undefined));
+    assert.deepEqual([undefined], transform([undefined]));
```

Comment:
maybe also add one for null.

Reply:
+1

Reply:
Thanks for suggesting more tests - I found out that there was also a problem with {a:undefined}. To solve that, I needed to add another traversal :/